business/istio_validations.go

```diff
@@ -480,6 +485,102 @@ func (in *IstioValidationsService) fetchDetails(rValue *kubernetes.IstioDetails,
 	}
 }
 
+func (in *IstioValidationsService) fetchExportedResources(exportedResources *kubernetes.ExportedResources, namespace string, errChan chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	if len(errChan) > 0 {
+		return
+	}
+	nss, err := in.businessLayer.Namespace.GetNamespaces()
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	currentVSs, err := in.fetchVirtualServices(namespace)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	currentDRs, err := in.fetchDestinationRules(namespace)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	currentSEs, err := in.fetchServiceEntries(namespace)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	for _, ns := range nss {
+		vsList, err := in.fetchVirtualServices(ns.Name)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		if containsExportToNamespacesIstioObjects(ns, currentVSs) {
+			exportedResources.VirtualServices = append(exportedResources.VirtualServices, vsList...)
+		}
+
+		drList, err := in.fetchDestinationRules(ns.Name)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		if containsExportToNamespacesIstioObjects(ns, currentDRs) {
+			exportedResources.DestinationRules = append(exportedResources.DestinationRules, drList...)
+		}
+
+		seList, err := in.fetchServiceEntries(ns.Name)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		if containsExportToNamespacesIstioObjects(ns, currentSEs) {
+			exportedResources.ServiceEntries = append(exportedResources.ServiceEntries, seList...)
+		}
+	}
+}
+
+func containsExportToNamespacesIstioObjects(ns models.Namespace, currentIstioObjects []kubernetes.IstioObject) bool {
+	for _, vs := range currentIstioObjects {
+		if exportToSpec, found := vs.GetSpec()["exportTo"]; found {
+			if namespaces, ok := exportToSpec.([]interface{}); ok {
+				for _, exportToNs := range namespaces {
+					// take only namespaces where it is exported to, or if it is exported to current namespace
+					if (exportToNs == "." && ns.Name == vs.GetObjectMeta().Namespace) || exportToNs == ns.Name || exportToNs == "*" {
```

Comment:
What if the IstioObject doesn't have the `exportTo` field?
Theoretically, when it doesn't exist it means that is exported to all the namespaces[0].

[0] https://istio.io/latest/docs/reference/config/networking/virtual-service/#VirtualService